src/MDI/mdi_plugin.cpp

```diff
@@ -74,18 +74,23 @@ MDIPlugin::MDIPlugin(LAMMPS *_lmp, int narg, char **arg) : Pointers(_lmp)
 
   // error checks
 
-  if (!mdi_arg || !infile_arg || !lammps_command)
-    error->all(FLERR, "MDI plugin must specify mdi, infile, command keywords");
+  if (!mdi_arg || !lammps_command)
+    error->all(FLERR, "MDI plugin must specify mdi and command keywords");
 
   // build full plugin_args string for args to plugin library
 
-  int n = strlen(mdi_arg) + strlen(infile_arg) + strlen(extra_arg) + 16;
```

Comment:
This kind of thing is much cleaner and simpler done by assembling the command line string(s) in a std::string variable using `fmt::format()` and then using utils::strdup to make a dynamically allocated copy.